license-maven-plugin/src/main/java/com/mycila/maven/plugin/license/AbstractLicenseMojo.java

```diff
@@ -492,13 +500,29 @@ protected final void execute(final Callback callback) throws MojoExecutionExcept
       }
 
       // make default base dir canonical
-      this.defaultBasedir = this.getCanonicalFile(this.defaultBasedir, "license.basedir");
+      workspace.basedir = getCanonicalFile(firstNonNull(workspace.basedir, legacyDefaultBasedir), "license.workspace.basedir");
```

Comment:
I would set `workspace.basedir` like this:

`workspace.basedir = getCanonicalFile(workspace.basedir, "license.workspace.basedir");`

but then `if(workspace.basedir == null)`, I would look for the `legacyDefaultBasedir`. And if this one is null, I would use the project.basedir.

The idea is to make the new config always a priority over the legacy one, the legacy one being used only as a fallback.